homeassistant/components/eight_sleep/binary_sensor.py

```diff
@@ -18,22 +26,23 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 
     name = "Eight"
     sensors = discovery_info[CONF_BINARY_SENSORS]
-    eight = hass.data[DATA_EIGHT]
+    eight = hass.data[DATA_EIGHT][DATA_API]
+    heat_coordinator = hass.data[DATA_EIGHT][DATA_HEAT]
 
     all_sensors = []
 
     for sensor in sensors:
-        all_sensors.append(EightHeatSensor(name, eight, sensor))
+        all_sensors.append(EightHeatSensor(name, heat_coordinator, eight, sensor))
 
     async_add_entities(all_sensors, True)
```

Comment:
We don't need to call update if the coordinator is already updated before platform setup.

Reply:
I will address this in my currently open PR